crates/oxc_parser/src/ts/types.rs

```diff
@@ -1382,7 +1382,11 @@ impl<'a> ParserImpl<'a> {
             kind if kind.is_update_operator() => true,
             Kind::LAngle | Kind::Await | Kind::Yield | Kind::Private | Kind::At => true,
             kind if kind.is_binary_operator() => true,
-            kind => kind.is_identifier(),
+            kind => {
+                kind.is_identifier_reference(self.ctx.has_yield(), self.ctx.has_await())
+                    && !kind.is_strict_mode_contextual_keyword()
+                    && !kind.is_contextual_keyword()
+            }
```

Comment:
The `is_identifier` implementation in TypeScript is here https://github.com/microsoft/TypeScript/blob/15392346d05045742e653eab5c87538ff2a3c863/src/compiler/parser.ts#L2316-L2335.

Our implementation only excludes reserve keywords. Do we need to add an `is_ts_identifier` method to cover TypeScript implementation?

Reply:
Yes please, we need to get the intention across.